src/main/java/com/hida/Minter.java

```diff
@@ -0,0 +1,799 @@
+        //} else {
+           // Logger.error("Error with TokenType of: "+TokenType);
+            //throw new BadParameterException(TokenType, "TokenType");
+        //}
+
+        // assign base map the appropriate values
+        this.BaseMap.put(TokenType.DIGIT, DIGIT_TOKEN);
+        if (sansVowel) {
+            this.BaseMap.put(TokenType.LOWERCASE, SANS_VOWEL_TOKEN);
+            this.BaseMap.put(TokenType.UPPERCASE, SANS_VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put(TokenType.MIXEDCASE, 
+                    SANS_VOWEL_TOKEN + SANS_VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put(TokenType.LOWER_EXTENDED, DIGIT_TOKEN + SANS_VOWEL_TOKEN);
+            this.BaseMap.put(TokenType.UPPER_EXTENDED, 
+                    DIGIT_TOKEN + SANS_VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put(TokenType.MIXED_EXTENDED,
+                    DIGIT_TOKEN + SANS_VOWEL_TOKEN + SANS_VOWEL_TOKEN.toUpperCase());
+        } else {
+            this.BaseMap.put(TokenType.LOWERCASE, VOWEL_TOKEN);
+            this.BaseMap.put(TokenType.UPPERCASE, VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put(TokenType.MIXEDCASE, VOWEL_TOKEN + VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put(TokenType.LOWER_EXTENDED, DIGIT_TOKEN + VOWEL_TOKEN);
+            this.BaseMap.put(TokenType.UPPER_EXTENDED, DIGIT_TOKEN + VOWEL_TOKEN.toUpperCase());
+            this.BaseMap.put(TokenType.MIXED_EXTENDED,
+                    DIGIT_TOKEN + VOWEL_TOKEN + VOWEL_TOKEN.toUpperCase());
+        }
+        Logger.info("BaseMap values set to: "+BaseMap);
+    }
+
+    /**
+     * Constructor for CustomMinters. In a CustomMinter scheme, the generation of ids
+     * is dependent on the parameter CharMap. The CustomMinter will generate as many
+     * characters as the length of CharMap. Each letter of the CharMap dictates the
+     * range of characters in it's position.
+     *
+     * @param DatabaseManager Used to add and check ids against database.
+     * @param CharMap The mapping used to describe range of possible characters
+     * at each of the id's root's digits
+     * @param Prepend Designates the format of the id. Will not appear in
+     * database.
+     * @param Prefix The string that will be at the front of every id
+     * @param sansVowel Designates whether or not the id's root contains vowels.
+     * If the root does not contain vowels, the sansVowel is true; false
+     * otherwise.
+     * @throws BadParameterException thrown whenever a malformed or invalid
+     * parameter is passed
+     */
+    public Minter(DatabaseManager DatabaseManager, String Prepend, String CharMap, String Prefix,
+            boolean sansVowel) throws BadParameterException {
+        this.DatabaseManager = DatabaseManager;
+        this.Prepend = Prepend;
+        this.Prefix = Prefix;
+        this.RootLength = CharMap.length();
+        this.SansVowel = sansVowel;
+        this.TokenType = convertToToken(CharMap);

```

Comment:
Just take a list/array of `TokenType` then you don't have to worry about an arbitrary string containing invalid characters